File: pretty_price.py
#Entendí mal el ejercicio

def pretty_price(bruto, margin):
    if margin == int(margin):
        margin = '0.' + str(margin)
        margin = float(margin)
        return bruto + margin
    elif margin == float(margin):
        return bruto + margin
  

print(pretty_price(3.44, 0.55))
print(pretty_price(3.44, 55))
print(pretty_price(3.44, .55))
print(pretty_price(3, 2))



#Se buscaba que el resultado fuera 3.95 y que se le pasara, pretty_price(3.5, 0.95) y diera ese resultado.

#Solución Jordan

def pretty_price(gross_price, extension):
    if isinstance(extension, int):
        extension = extension * 0.01

    return int(gross_price) + extension


print(pretty_price(3.20, 99))
print(pretty_price(3.99, 0.20))
print(pretty_price(3.20, .95))
print(pretty_price(3, 95))
print(pretty_price(3, 2))

File: prettyPrice.js
const prettyPrice = (total, extension) => {
    return extension === parseInt(extension) ? Math.floor(total) + extension * 0.01 : Math.floor(total) + extension
}


console.log(prettyPrice(3.45, 95));


/*Solución Jordan:
const prettyPrice = (grossPrice, extension) => {
  if (Number.isInteger(extension)) {
    extension = extension * 0.01;
  }
  return Math.floor(grossPrice) + extension;
};

prettyPrice(2.2, 0.95);
prettyPrice(2.2, 95);
prettyPrice(2.2, 0); 
prettyPrice(2.2, 99);
*/